haystack/components/converters/openapi_functions.py

```diff
@@ -176,22 +216,26 @@ def _parse_openapi_spec(self, content: str) -> Dict[str, Any]:
         :return: The parsed OpenAPI specification.
         :rtype: Dict[str, Any]
         """
+        open_api_spec_content = None
         try:
-            return json.loads(content)
+            open_api_spec_content = json.loads(content)
         except json.JSONDecodeError as json_error:
             # heuristic to confirm that the content is likely malformed JSON
             if content.strip().startswith(("{", "[")):
                 raise json_error
 
         try:
-            return yaml.safe_load(content)
+            open_api_spec_content = yaml.safe_load(content)
         except yaml.YAMLError:
             error_message = (
                 "Failed to parse the OpenAPI specification. "
                 "The content does not appear to be valid JSON or YAML.\n\n"
             )
             raise RuntimeError(error_message, content)
 
+        # Replace references in the object with their resolved values, if any
```

Comment:
This step is necessary to resolve all the refs in the newly loaded schema. Complete schema types will replace all the ` $ref` instances in the schema. It works for both YAML and JSON specs - confirmed in a separate offline test.